File: libs/server/domain/src/domain/workspace/workspaces.ts
import { HasMany } from '../core';
import { Workspace, WorkspaceDescription } from './workspace';

export interface Workspaces extends HasMany<Workspace> {
  create(
    ownerUserId: string,
    description: WorkspaceDescription,
  ): Promise<Workspace>;
  update(id: string, description: WorkspaceDescription): Promise<Workspace>;
  delete(id: string): Promise<void>;
}

File: libs/server-java/domain/src/main/java/reengineering/ddd/evidence/domain/model/Workspaces.java
package reengineering.ddd.evidence.domain.model;

import io.github.jayclock.smartdomain.core.HasMany;
import reengineering.ddd.evidence.domain.description.WorkspaceDescription;

public interface Workspaces extends HasMany<String, Workspace> {
  Workspace create(String ownerUserId, WorkspaceDescription description);

  Workspace update(String workspaceId, WorkspaceDescription description);

  void delete(String workspaceId);
}
